cdc/sink/dispatcher/expression.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dispatcher
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/pingcap/tiflow/pkg/errors"
+)
+
+var topicNameRE = regexp.MustCompile(`^[A-Za-z0-9\._\-]*\{schema\}[A-Za-z0-9\._\-]*$|^\{schema\}_\{table\}$`)
+var forbidRE = regexp.MustCompile(`[^a-zA-Z0-9\._\-]`)
+var schemaRE = regexp.MustCompile(`\{schema\}`)
+var tableRE = regexp.MustCompile(`\{table\}`)
+
+// Parse convert schema/table name to kafka topic name by using a topic expression.
+// A valid kafka topic name matches [a-zA-Z0-9\._\-]{1,249},
+// and the topic expression only accepts two types of expressions:
+// 	1. [prefix]{schema}[suffix], the prefix/suffix is optional and matches [A-Za-z0-9\._\-]*
+//  2. {schema}_{table}
+
+// When doing conversion, the special characters other than [A-Za-z0-9\._\-] in schema/table
+// will be substituted for underscore '_'
+func Parse(topicExpr, schema, table string) (string, error) {
```

Comment:
This function seems better named `Substitute` or `GeneratorTopic`

Reply:
The function name is changed.